src/narjure/memory_management/concept_creator.clj

```diff
@@ -1,28 +1,44 @@
 (ns narjure.memory-management.concept-creator
   (:require
-    [co.paralleluniverse.pulsar.actors :refer [! spawn gen-server register! cast! Server self shutdown! unregister! set-state! state]]
+    [co.paralleluniverse.pulsar.actors
+     :refer [! spawn gen-server register! cast! Server self
+             shutdown! unregister! set-state! state whereis]]
     [narjure.memory-management.concept :refer [concept]]
+    [narjure.memory-management.task-dispatcher :refer [c-map]]
     [narjure.actor.utils :refer [defactor]]
     [taoensso.timbre :refer [debug info]])
   (:refer-clojure :exclude [promise await]))
 
 (def aname :concept-creator)
+(def max-concepts 1000)
 
 (defn create-concept
-  ;TODO: update state for concept-actor to state initialiser
-  ;TODO: Create required sub-term concepts
+  "Create a concept for the supplied term in c-map and
+   add a key value pair to c-map for the created concept.
+   Updates the concept count and if it the count exceeds
+   the specified limit a :concept-limit-msg is posted to
+   :forgettable-concept-collator"
   [term c-map]
-  (swap! c-map assoc term (spawn concept))
-  #_(debug aname (str "Created concept: " term))
-  )
+  (let [concept-ref (spawn (concept))]
+    (swap! c-map assoc term concept-ref)
+    (set-state! (update @state :concept-count inc))
+    (cast! concept-ref [:set-content-msg term]))
+  (if (> (:concept-count @state) max-concepts)
+    (cast! (:forgettable-concept-collator @state) [:concept-limit-msg]))
+  #_(debug aname (str "Created concept: " term)))
 
 (defn task-handler
-  "When concept-map does not contain :term, create concept actor for term
-   then post task to task-dispatcher either way."
+  "Create a concept for each term in statement plus one for
+   occurrence time if they dont exist. Then post the task
+   back to task-dispatcher. Note that occurrence time keys
+   are keys, whilst term keys are terms, which can be numbers"
   [from [msg task c-map]]
```

Comment:
It could be something like 
```
  [from [_ {:keys [statement occurrence] :as task} c-map]]
  (doseq [term (conj (:terms statement) occurrence)]
    (when-not (@c-map term)
      (create-concept term c-map)))
```
We can avoid duplication of `create-concept` invocation, replace `if-not` by `when-not` as "else" branch is not used here, and move desctructuring of task to function's parameters declaration.